packages/plugin-legacy/index.js

```diff
@@ -559,6 +565,44 @@ function replaceLegacyEnvBabelPlugin() {
   })
 }
 
+/**
+ * @param {any} targets
+ * @param {boolean} ignoreBrowserslistConfig
+ * @returns {'defaults'|undefined}
+ */
+function getDefaultTargets(targets, ignoreBrowserslistConfig) {
+  if (targets) return targets
+
+  if (ignoreBrowserslistConfig) {
+    return defaultTargets
+  }
+
+  try {
+    // fix https://github.com/vitejs/vite/issues/2476
+    // @babel/preset-env automatically detects browserslist config sources:
+    // browserslist field in package.json
+    // .browserslistrc file in cwd.
+
+    const getRealPath = (dir) => path.posix.join(process.cwd(), dir)
+    const pkgStr = fs.readFileSync(getRealPath('package.json'), 'utf8')
+    const pkgJson = JSON.parse(pkgStr)
+
+    // if the browserslist field in package.json
+    if (pkgJson.browserslist) {
+      return undefined
+    }
+
+    // if the .browserslistrc file in cwd.
+    const content = fs.readFileSync(getRealPath('.browserslistrc'), 'utf8')
+    if (content) {
+      return undefined
+    }
+    return defaultTargets
+  } catch (error) {
+    return defaultTargets
```

Comment:
**suggestion (non-blocking):** Instead of returning `defaultTargets` in try and catch, you could return it in a `finally`

```js
  // ...
    if (content) return undefined
  } finally {
    return defaultTargets
```

Reply:
This should be wrong, it will always return the finally value